solaris/SolarisProcessList.c

```diff
@@ -50,13 +52,18 @@ ProcessList* ProcessList_new(UsersTable* usersTable, Hashtable* pidMatchList, ui
 
    spl->kd = kstat_open();
 
-   pl->cpuCount = sysconf(_SC_NPROCESSORS_ONLN);
+   pageSize = sysconf(_SC_PAGESIZE);
+   if (pageSize == -1)
+      CRT_fatalError("Cannot get pagesize by sysconf(_SC_PAGESIZE)");
+   pageSizeKB = pageSize / 1024;
 
-   if (pl->cpuCount == 1 ) {
+   pl->cpuCount = sysconf(_SC_NPROCESSORS_ONLN);
+   if (pl->cpuCount == -1)
```

Comment:
Why not check `pl->cpuCount < 1`? Zero CPUs ain't that helpful. ;-)

Reply:
Heh, indeed.  It's a matter of following the docs exactly I guess (I'm just keeping the existing code convention going there) - the syscall is documented to return -1 on failure.  If zero was returned the syscall would be broken of course - by definition, at least one CPU is always online: the one currently executing htop. ;)  Worse, if we treated that as an "error" the call to CRT_fatalError uses errno which presumably would not have been set since zero is not an error (even though semantically it is an error of course).